pkg/diagnose/cni.go

```diff
@@ -206,3 +222,93 @@ func mustHaveSubmariner(clusterInfo *cluster.Info) {
 		panic("cluster.Info.Submariner field cannot be nil")
 	}
 }
+
+func checkOVNVersion(info *cluster.Info, status reporter.Interface) bool {
+	status.Start("Checking OVN version")
+	defer status.End()
+
+	clientSet := info.ClientProducer.ForKubernetes()
+
+	ovnPod, err := findPod(clientSet, ovnKubeDBPodLabel)
+	if err != nil || ovnPod == nil {
+		status.Failure("Failed to get OVNKubeDB Pod %v", err)
+		return false
+	}
+
+	ovnNBVersion, err := getOVNNBVersion(clientSet, info.RestConfig, ovnPod)
+	if err != nil {
+		status.Failure("Failed to get OVN NB version %v", err)
+	}
+
+	if err != nil {
+		status.Failure("Failed to get OVN NB version %v", err)
+		return false
+	}
+
+	if version.Compare(ovnNBVersion, minOVNNBVersion, "<") {
+		status.Failure("OVN NBDB version %v is lesser than minimum supported %v", ovnNBVersion, minOVNNBVersion)
+		return false
+	}
+
+	status.Success("OVN NBDB Version %v is supported", ovnNBVersion)
+
+	return true
+}
+
+func findPod(clientSet kubernetes.Interface, labelSelector string) (*corev1.Pod, error) {
+	pods, err := clientSet.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{
+		LabelSelector: labelSelector,
+		Limit:         1,
+	})
+	if err != nil {
+		return nil, errors.WithMessagef(err, "error listing Pods by label selector %q", labelSelector)
+	}
+
+	if len(pods.Items) == 0 {
+		return nil, errors.WithMessagef(err, "found 0 pods with label %v", labelSelector)
+	}
+
+	return &pods.Items[0], nil
+}
+
+func getOVNNBVersion(clientSet kubernetes.Interface, config *rest.Config, pod *corev1.Pod) (string, error) {
+	// We can run ovn-nbctl in any of the containers in db pod
```

Comment:
No DB pod downstream, I'd be sure to get in the nbdb container to be safe

Reply:
Problem is the name of nbdb container varies between vanilla OVNK [used in kind] and OCP. Not sure about downstream, but I tested with upstream version. These are the containers in `ovnkube-master` pod

`northd, nbdb, kube-rbac-proxy, sbdb, ovnkube-master, ovn-dbchecker`

`ovn-nbctl` works in all except for `kube-rbac-proxy`. A cleaner way would be to go through container names and pick the nbdb one.